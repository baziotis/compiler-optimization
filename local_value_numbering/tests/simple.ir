.0:
  %0 <- 0
  %1 <- 1
  %2 <- 2
  %3 <- 3
  BR .1		

.1:
  %0 <- %1 + %2
  %1 <- %0 + %3
  %2 <- %1 + %2
  %4 <- %0 + %3
  %4 <- %3
  %5 <- %0 + %4
  BR .2

.2:
  %0 <- 17
  %1 <- 17 + 18
  %2 <- %0 + 18
