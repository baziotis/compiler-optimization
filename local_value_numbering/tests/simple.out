Number of BBs: 3
.0:                         ;; preds:  --  succs: 1
  %0 <- 0
  %1 <- 1
  %2 <- 2
  %3 <- 3
  BR .1		

.1:                         ;; preds: 0 --  succs: 2
  %0 <- %1 + %2
  %1 <- %0 + %3
  %2 <- %1 + %2
  %4 <- %1
  %4 <- %3
  %5 <- %1
  BR .2		

.2:                         ;; preds: 1 --  succs: 
  %0 <- 17
  %1 <- 17 + 18
  %2 <- %1

